### PR TITLE
MMH-124 Add in-house protein database support

### PR DESCRIPTION
**Pull request by AlexRajya**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,6 +52,9 @@ Note that these match the 2024.2 Desktop release.
 
 ## In-house Database Configuration
 
+:warning: **OnSite WebCSD makes the assumption structure identifiers are unique across all registered in-house databases.
```
Comment (aseidu-ccdc) [inline README.md:55]: I think "OnSite WebCSD assumes"...may be clearer

File: docker-compose.yml [hunk 2]
```diff
@@ -15,189 +18,287 @@ services:
...
-    image: ccdcrepository.azurecr.io/onsite/ccdc-csd-fingerprint-service:4.0.0
+    image: ccdcrepository.azurecr.io/onsite/ccdc-csd-fingerprint-service:4.2.0
     restart: unless-stopped
+    logging:
+      driver: local
 
   ccdc-csd-substructure-filter:
     labels:
       <<: *k8s-labels
     image: ccdcrepository.azurecr.io/onsite/ccdc-csd-substructure-filter-service:4.0.0
```
Comment (aseidu-ccdc) [inline docker-compose.yml:239]: Do the services from here downwards not need an  image update?

**Review by aseidu-ccdc — APPROVED** [review]

File: docker-compose.yml [hunk 3]
```diff
@@ -15,189 +18,287 @@ services:
...
+      start_period: 30s
+    logging:
+      driver: local
 
   ccdc-file-generation:
     restart: unless-stopped
     image: ccdcrepository.azurecr.io/onsite/ccdc-structure-file-generation:4.1.0
     environment:
-      - CacheSettings__ConnectionString=${DB_CONNECTIONSTRING};Database=structure_file_cache
+      - CacheSettings__ConnectionString=${DB_CONNECTIONSTRING};Database=testsuite_structure_file_cache
```
Comment (flintd) [inline docker-compose.yml:290]: Just checking this should be test suite_*

File: README.md [hunk 4]
```diff
@@ -63,6 +66,27 @@ To enable these databases follow:
 
 More information is given in the notes & example sections of the sample file. This acts as an [override file](https://docs.docker.com/compose/extends/) which you will have to include in the startup command.
 
+### In-house Protein Database Creation and Configuration
+
+If you have a Macromolecule Hub licence, you can also create and register in-house protein databases for use within OnSite WebCSD.
+
+To create an In-house Protein Database, you will need the Python Utilities. These can be downloaded from the [CCDC Downloads page](https://www.ccdc.cam.ac.uk/support-and-resources/csdsdownloads/)
+under `CSD Python API > Python CSD Python API Utilities`. If you do not see the CSD Python API listed in the available downloads, you may need to sign in.
+Relevant script can be found under `ccdc > utitities > create_protein_database`.
```
Comment (flintd) [inline README.md:75]: The relevant script...

**Review by flintd — APPROVED** [review]

File: docker-compose.yml (hunk 2 above)
Comment (AlexRajya) [reply docker-compose.yml:239]: No these haven't had any changes so no need to bump the version. We used to update them anyway but people made the choice not to do that anymore

File: docker-compose.yml (hunk 3 above)
Comment (AlexRajya) [reply docker-compose.yml]: Good spot!

**Pull request merged**